Breadcrumbs.py

```diff
@@ -108,4 +141,223 @@ def get_points_by_row(row):
+
+  def close(self):
+    self.view.erase_phantoms('breadcrumbs')
+    self.phantoms_visible = False
+
+  def navigate(self, href, string):
+    if href == 'close':
+      self.close()
+    else:
+      copy(self.view, string)
+
+  def run(self, edit):
+    if self.phantoms_visible:
+      self.close()
+      return
+
+    stylesheet = '''
+      <style>
+        html {
+          --base-bg: color(var(--bluish) blend(var(--background) 30%));
+          --accent-bg: color(var(--base-bg) blend(var(--foreground) 90%));
+          line-height: 20px;
+        }
+        div.phantom {
+          margin: 0;
+        }
+        i,
+        .crumb {
+          line-height: 2em;
+          padding-right: 6px;
+        }
+        i {
+          padding-left: 6px;
+        }
+        .separator {
+          border: 1em solid;
+          width: 0;
+          height: 0;
+          font-size: inherit;
+          line-height: 0px;
+        }
+        i,
+        .crumb,
+        .separator {
+          background-color: var(--base-bg);
+        }
+        .crumb-2,
+        .separator-2 {
+          background-color: var(--accent-bg);
+        }
+        .separator-1 {
+          border-color: var(--base-bg);
+          border-left-color: var(--accent-bg);
+        }
+        .separator-2 {
+          border-color: var(--accent-bg);
+          border-left-color: var(--base-bg);
+        }
+        div.phantom a {
+          text-decoration: inherit;
+          vertical-align: middle;
+          line-height: 2em;
+          padding: 0 7px 0 12px;
+          background-color: var(--base-bg);
+        }
+        div.phantom a.close {
+          font-weight: bold;
+          padding-left: 4px;
+        }
+      </style>
+    '''
+
+    template = '''
+      <body id="inline-breadcrumbs">
+        {stylesheet}
+        <div class="phantom">{breadcrumbs}<a href="copy">Copy</a><a class="close" href="close">''' + chr(0x00D7) + '''</a></div>
+      </body>
+    '''
+
+    phantoms = []
+
+    for region in self.view.sel():
+      (row, col) = self.view.rowcol(region.begin())
+
+      crumb_elements = []
+      breadcrumbs = make_breadcrumbs(self.view, row)
+      breadcrumbs_string = get_separator(self.view).join(breadcrumbs)
+
+      if len(breadcrumbs) > 0:
+        for i, crumb in enumerate(breadcrumbs):
+          parity = (i % 2) + 1
+          crumb_elements.append('<span class="separator separator-{parity}"> </span><span class="crumb crumb-{parity}">'.format(parity=parity) + html.escape(crumb, quote=False) + '</span>')
+        breadcrumbs_element = ''.join(crumb_elements)
+      else:
+        breadcrumbs_element = '<i>None</i>'
+
+      body = template.format(
+          breadcrumbs=breadcrumbs_element,
+          stylesheet=stylesheet
+      )
+      phantom = sublime.Phantom(
+          region,
+          body,
+          sublime.LAYOUT_BLOCK,
+          on_navigate=lambda href,
+          string=breadcrumbs_string: self.navigate(href, breadcrumbs_string)
```

Comment:
> So, I guess this code doesn't really work as you think, when there is more than one cursor: it will always copy the text from the last occurrence.

Yeah, was definitely missing some of how this works. It's nasty, because it looks like it works until you have more than one cursor.